### PR TITLE
Update contributing

### PR DESCRIPTION
**Pull request by MonicaOlejniczak**
# ↪️ Pull Request

This change restructures the contribution page in more sections to make the content clearer. In addition to this, I have:
* Added debugging information to the contribution development workflow section
* Added recommended VSCode extensions, this will create a popup in the GUI asking for users to install extensions they do not have
* Added a new VSCode launch configuration for running integration tests with lldb, unfortunately I couldn't get llnode working even with the exact config posted in Discord previously
* Reordered the gitignore (mostly alphabetically by dir then file, as it appear in file tree explorers)

## 💻 Examples

N/A

## 🚨 Test instructions

Compare [previous](https://github.com/parcel-bundler/parcel/blob/v2/CONTRIBUTING.md) with [new](https://github.com/parcel-bundler/parcel/blob/molejniczak/contributing/CONTRIBUTING.md)

### CODE REVIEW

**Review by yamadapc — APPROVED** [review]

**Review by JakeLane — APPROVED** [review]

File: .vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  // See https://go.microsoft.com/fwlink/?LinkId=827846 to learn about workspace recommendations.
+  "recommendations": [
+    "dbaeumer.vscode-eslint",
+    "eamodio.gitlens",
```
Comment (mischnic) [inline .vscode/extensions.json]:
```suggestion
```

GitLens is really depends on personal preference and is not required by the project. I don't use it for example

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -10,57 +10,80 @@ Have a question or feature request? Rather than opening an issue, use the [Discu
...
 
-In order to make it easier to get familiar with the codebase we labeled simpler issues using [Good First Issue](https://github.com/parcel-bundler/parcel/issues?q=is%3Aopen+is%3Aissue+label%3A%22%E2%9C%A8+Parcel+2%22+label%3A%22%3Ababy%3A+Good+First+Issue%22) and [Help Wanted](https://github.com/parcel-bundler/parcel/issues?q=is%3Aopen+is%3Aissue+label%3A%22%E2%9C%A8+Parcel+2%22+label%3A%22%3Apray%3A+Help+Wanted%22).
+Before starting make sure you have the following installed:
 
-Before starting make sure you have the following requirements installed: [git](https://git-scm.com), [Node](https://nodejs.org), [Yarn](https://yarnpkg.com) and [Rust](https://www.rust-lang.org/tools/install).
+- [git](https://git-scm.com)
+- [Node](https://nodejs.org) at LTS
+- [Yarn](https://yarnpkg.com) at v1
+- [Rust](https://www.rust-lang.org/tools/install) stable
+- [Flow](https://flow.org/en/docs/editors) IDE autocompletion and type-checking
```
Comment (mischnic) [inline CONTRIBUTING.md:21]: You don't have to install flow separately though, it's installed via a npm dependency?

Comment (MonicaOlejniczak) [reply CONTRIBUTING.md:21]: It's for the IDE integrations, it was previously mentioned in the file as

> Parcel uses [Flow](https://flow.org/) for type checking. If you're using an IDE, make sure to install the [Flow extension](https://flow.org/en/docs/editors/) to ensure your editor's autocomplete and type-checking works as expected.

Let me know if you still want it there, whether the wording should be updated, or removed entirely

File: .vscode/extensions.json (hunk 1 above)
Comment (MonicaOlejniczak) [reply .vscode/extensions.json]: Sure, fixed

**Review by mischnic — APPROVED** [review]

**Comment by Aaaaaaaaaaadeel** [comment]
Error: The specified module could not be found.        
\\?\C:\Users\A.S Computers\Desktop\react\node_modules\@parcel\source-map\parcel_sourcemap_node\artifacts\index.win32-x64-msvc.node
    at Object..node (node:internal/modules/cjs/loader:1921:18)
    at Module.load (node:internal/modules/cjs/loader:1465:32)
    at Function._load (node:internal/modules/cjs/loader:1282:12)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:235:24)
    at Module.require (node:internal/modules/cjs/loader:1487:12)
    at require (node:internal/modules/helpers:135:16)  
    at Object.<anonymous> (C:\Users\A.S Computers\Desktop\react\node_modules\@parcel\source-map\parcel_sourcemap_node\index.js:15:18)
    at Module._compile (node:internal/modules/cjs/loader:1730:14)
    at Object..js (node:internal/modules/cjs/loader:1895:10) {
  code: 'ERR_DLOPEN_FAILED'
} help me to solve this error

**Pull request merged**